src/functionalTest/java/io/knotx/stack/functional/FragmentsDebugDataTest.java

```diff
@@ -107,22 +114,39 @@ void requestPage(VertxTestContext testContext, Vertx vertx,
       "scenarios/fragments-debug-data/mocks.conf",
       "scenarios/fragments-debug-data/tasks.conf"})
   void requestWebApi(VertxTestContext testContext, Vertx vertx,
+      @RandomPort Integer brokenServicePort, @RandomPort Integer delayedServicePort,
       @RandomPort Integer globalServerPort) {
+    // given
+    givenBrokenServiceServer(brokenServicePort);
+    givenDelayedServiceServer(delayedServicePort);
     givenServerTester(globalServerPort);
+
+    // when & then
     knotxShouldAppendConsumerDataToFragmentBody(testContext, vertx);
   }
 
   private void givenServerTester(Integer globalServerPort) {
     serverTester = KnotxServerTester.defaultInstance(globalServerPort);
   }
 
-  private void givenDelayedServiceServer(Integer delayedServicePort) {
-    delayedServiceServer = new WireMockServer(delayedServicePort);
-    delayedServiceServer.stubFor(get(urlEqualTo("/mock/scenario/delayed")).willReturn(
-        aResponse()
-            .withStatus(200)
-            .withFixedDelay(200)));
-    delayedServiceServer.start();
+  private void givenBrokenServiceServer(Integer port) {
+    brokenServiceMock = new WireMockServer(port);
+    brokenServiceMock.stubFor(
+        get(urlEqualTo("/mock/scenario/broken"))
+            .willReturn(aResponse().withStatus(500)));
+    brokenServiceMock.stubFor(
+        post(urlEqualTo("/mock/scenario/analytics"))
+            .withRequestBody(equalToJson("{\"user-data-id\":\"3214232\"}"))
```

Comment:
Maybe you could export it to const

Reply:
It is string value in service configuration. I can not point from code so we can skip it.